src/app/tab3/register/register.component.ts

```diff
@@ -1,14 +1,53 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { RequestEnums } from 'src/app/shared/constants/request-enums';
+import { CommonRequestService } from 'src/app/shared/services/http/common-request.service';
 
 @Component({
   selector: 'app-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.scss'],
 })
 export class RegisterComponent implements OnInit {
+  registerForm: FormGroup;
+ 
+  validationMessages = {
+    username: [
+      {type: 'required', message: 'Please enter Email'},
+      {type: 'pattern', message: 'Please enter Valid Email'}
+    ],
+  };
+ 
+  constructor(private formBuilder: FormBuilder,
+    private commonRequestService: CommonRequestService) { }
 
-  constructor() { }
-
-  ngOnInit() {}
-
+  ngOnInit() {
+    this.initRegistration();
+  }
+  initRegistration() {
+    this.registerForm = this.formBuilder.group({
+      username: ['', [Validators.required, Validators.pattern("[a-zA-Z0-9.-_]{1,}@[a-zA-Z.-]{2,}[.]{1}[a-zA-Z]{2,}")]],
+      provider:['EMAIL'],
```

Comment:
Use this Hard coded values from the app-properties.enum file if exist else create a file inside shared/enums

Reply:
User validation.compose.

Reply:
Done